plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1087,29 +1087,133 @@ public void testCaseInsensitivePredicate()
+        // additive inverse
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey + (-nationkey) = 0"))
+                .matches("SELECT nationkey, name, regionkey FROM nation")
+                .isFullyPushedDown();
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey - nationkey = 0"))
+                .matches("SELECT nationkey, name, regionkey FROM nation")
+                .isFullyPushedDown();
 
-        // some databases calculate remainder instead of modulus when one of the values is negative
-        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % -nationkey = 2"))
-                .isFullyPushedDown()
-                .matches("VALUES (BIGINT '3', CAST('CANADA' AS varchar(25)), BIGINT '1')");
+        // constant
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey + 1 = 7"))
+                .matches("VALUES (BIGINT '6', CAST('FRANCE' AS varchar(25)), BIGINT '3')")
+                .isFullyPushedDown();
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey - 1 = 7"))
+                .matches("VALUES (BIGINT '8', CAST('INDIA' AS varchar(25)), BIGINT '2')")
+                .isFullyPushedDown();
+
+        // column reference
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey + regionkey = 26"))
+                .matches("VALUES (BIGINT '23', CAST('UNITED KINGDOM' AS varchar(25)), BIGINT '3')")
+                .isFullyPushedDown();
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE nationkey - regionkey = 23"))
+                .matches("VALUES (BIGINT '24', CAST('UNITED STATES' AS varchar(25)), BIGINT '1')")
+                .isFullyPushedDown();
+        // column reference with AND
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE regionkey = 0 AND nationkey + regionkey = 14"))
+                .matches("VALUES (BIGINT '14', CAST('KENYA' AS varchar(25)), BIGINT '0')")
+                .isFullyPushedDown();
+        assertThat(query("SELECT nationkey, name, regionkey FROM nation WHERE regionkey = 0 AND nationkey - regionkey = 16"))
+                .matches("VALUES (BIGINT '16', CAST('MOZAMBIQUE' AS varchar(25)), BIGINT '0')")
+                .isFullyPushedDown();
 
-        assertQueryFails("SELECT nationkey, name, regionkey FROM nation WHERE nationkey > 0 AND (nationkey - regionkey) % 0 = 2", errorMessageForDivisionByZeroPushdown());
+        // non-integer types shouldn't get pushed down
+        assertThat(query("SELECT custkey, name, acctbal FROM customer WHERE acctbal + custkey = 524.2"))
+                .isNotFullyPushedDown(FilterNode.class);
+        assertThat(query("SELECT custkey, name, acctbal FROM customer WHERE acctbal - custkey = 119.65"))
+                .isNotFullyPushedDown(FilterNode.class);
```

Comment:
At least for some connectors these SHOULD get pushed down.
Why are you being so negative about this?

Let's at least move this to `testFloatingPointArithmeticPredicatePushdown` so a better connector doesn't need to override this whole big method.

Reply:
Good point.
I added the test only to verify that the `integer_type` we define when creating the rewriter is respected.

I'll extract it to separate method but I don't think this test case serves any real purpose. For example we don't explicitly test everything that we don't support.

Reply:
removed this case.